src/lib.rs

```diff
@@ -11,15 +11,17 @@ use shared_socket::SocketHeld;
 
 // pyO3 module
 use pyo3::prelude::*;
-use types::{FunctionInfo, Response};
+use types::{function_info::FunctionInfo, request::PyRequest, response::PyResponse, Url};
 
 #[pymodule]
 pub fn robyn(_py: Python<'_>, m: &PyModule) -> PyResult<()> {
     // the pymodule class to make the rustPyFunctions available
     m.add_class::<Server>()?;
     m.add_class::<SocketHeld>()?;
     m.add_class::<FunctionInfo>()?;
-    m.add_class::<Response>()?;
+    m.add_class::<PyRequest>()?;
+    m.add_class::<PyResponse>()?;
+    m.add_class::<Url>()?;
```

Comment:
@AntoineRR , why do we need to export this?

Reply:
This is because we can then use the `Url` struct in python, and do things like `request.url.host` instead of `request.url["host"]`

Reply:
@AntoineRR , wouldn't that still work if you just export PyRequest ?

Reply:
I tested it and it does indeed work without exporting the `Url` struct. It probably makes no sense to allow the user to create `Url` instances from Python so I will remove this export.